api/v1alpha1/clusterextension_types.go

```diff
@@ -49,10 +49,9 @@ type ClusterExtensionSpec struct {
 	// packageName is a reference to the name of the package to be installed
 	// and is used to filter the content from catalogs.
 	//
-	// This field is required, immutable and follows the DNS label standard as defined in RFC
-	// 1123, with a deviation in the maximum length being no more than 48
-	// characters. This means that valid values:
-	//   - Contain no more than 48 characters
+	// This field is required, immutable and follows the DNS label standard as
+	// defined in [RFC 1123]. This means that valid values:
+	//   - Contain no more than 63 characters
```

Comment:
We had originally decided on a lower maximum so that we left a few characters for controllers and other clients to use packages names in their own objects with prefixes or suffixes totaling 15 characters.

Did we talk specifically about eliminating this cushion?

Reply:
Could you provide an example of a use case where the 48 character limit would be beneficial and actually be used? An additional note, the [package name validation](https://github.com/operator-framework/operator-registry/blob/2e0d6a237a1ae06998584043e44c44029f5df352/alpha/declcfg/declcfg_to_model.go#L26-L28) in `opm validate` uses the `validation.IsDNS1123Label` function implemented in https://github.com/kubernetes/kubernetes/blob/7436ca32bc766ff202109a7541d2e7bb41ee7d13/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L185-L202

which validates a maximum length of 63 characters. In your scenario, a package that has gone through `opm validate` may not be able to be installed due to the 48 character length limit we are imposing in the `ClusterExtension` API.

Reply:
The 48 character limit I think comes from the early days of rukpak. And the idea was originally because rukpak itself used that package name in order to create other objects on the cluster. So we decided to give ourselves and other rukpak provisioner developers 15 characters of slush in case they wanted to use package names to construct other object names.

That's a good point about `opm validate`. If we wanted to keep things at 48, we'd have to double check all known catalogs to see if that would break anything.